src/mountscript.sh

```diff
@@ -636,6 +636,16 @@ if ! is_valid_blob_fqdn "$nfs_host"; then
     exit 1
 fi
 
+#
+# Fail mount before resolution if the hostname entry is present in /etc/hosts
+#
+if is_present_in_etc_hosts "$nfs_host"; then
+    eecho "Mount failed!"
+    eecho "Detected entry for $nfs_host in /etc/hosts."
+    eecho "[Action Required]: Remove or comment out the entry for $nfs_host in /etc/hosts for MOUNT to work."
+    exit 1
+fi
```

Comment:
we need to be very careful that we don't raise a red flag, so let's first resolve the IP and then check for the entire
IP[[:space:]]+fqdn so that we only complain when user has set the resolver config such that the address is resolving from /etc/hosts.